week-2/Homework/extra/2-array-methods.js

```diff
@@ -5,31 +5,80 @@ Following are set of small functions that you need to create using loops and arr
 //1)Find the Smallest Number in an Array
 //Create a function that takes an array of numbers and return the smallest number in it
 
-function findSmallestNum(arr) {}
+function findSmallestNum(arr) {
 
-findSmallestNum([354, 15, 114, 2]); //Expected: 2
+    var smallestValue= Math.min.apply(Math,arr);
+    return smallestValue;
 
-findSmallestNum([-76, 1, -79, 1, 0]); //Expected: -79
+   }
+
+console.log(findSmallestNum([354, 15, 114, 2])); //Expected: 2
+
+console.log(findSmallestNum([-76, 1, -79, 1, 0])); //Expected: -79
 
 //2)Checking Even Numbers
 //Create a function that takes in an array and returns true if all its values are even, and false otherwise.
-function checkAllEven(arr) {}
+function checkAllEven(arr) {
+    for(var i=0;i<arr.length;i++){
+        if(arr[i]%2===0){
+            return true;
+        }
+        else{
+            return false;
+        }
+
+    }
+    
+}
 
-checkAllEven([1, 2, 2, 6, 9, 4]); //Expected: false
+console.log(checkAllEven([1, 2, 2, 6, 9, 4])); //Expected: false
 
-checkAllEven([2, 4, 6]); //Expected: true
+console.log(checkAllEven([2, 4, 6])); //Expected: true
 //3)Half, Quarter and Eighth
 //Create a function that takes a number and return an array of three numbers: half of the number, quarter of the number and an eighth of the number.
-function getHalfQuarterEighth(number) {}
-getHalfQuarterEighth(6); //Expected:[3, 1.5, 0.75]
+function getHalfQuarterEighth(number) {
+var newArr=[];
+var newArr2=number/2;
+var newArr4=number/4;
+var newArr8=number/8;
+newArr.push(newArr2);
+newArr.push(newArr4);
+newArr.push(newArr8);
+
+return newArr;
+
+
+}
+console.log(getHalfQuarterEighth(6)); //Expected:[3, 1.5, 0.75]
 
 //4)Add the index to the number
 //Given an array of numbers, create a function which returns the same array but with each element's index in the array added to itself. This means you add 0 to the number at index 0, add 1 to the number at index 1, etc...
-function addIndexes(arr) {}
-addIndexes([1, 4, 3, 4, 5]); //Expected: [1, 5, 5, 7, 9]
+function addIndexes(arr) {
+    var indexArray=[];
+     for(i=0;i<arr.length;i++){
+         var indexArray=arr[i]+arr.indexOf();

```

Comment:
You are re-assigning `indexArray` on each stage of your loop.
You need to ensure that you are pushing an expression into your array - you are close here should be something like:
`indexArray.push(arr[i] + ?)`
What can you replace `?` with in order to add the `index` to the array element `array[i]` each time